paasta_tools/setup_prometheus_adapter_config.py

```diff
@@ -228,6 +231,119 @@ def should_create_piscina_scaling_rule(
     return False, "did not request piscina autoscaling"
 
 
+def should_create_active_requests_scaling_rule(
+    autoscaling_config: AutoscalingParamsDict,
+) -> Tuple[bool, Optional[str]]:
+    """
+    Determines whether we should configure the prometheus adapter for a given service.
+    Returns a 2-tuple of (should_create, reason_to_skip)
+    """
+    if autoscaling_config["metrics_provider"] == "active-requests":
+        return True, None
+    return False, "did not request active-requests autoscaling"
+
+
+def create_instance_active_requests_scaling_rule(
+    service: str,
+    instance: str,
+    autoscaling_config: AutoscalingParamsDict,
+    paasta_cluster: str,
+    namespace: str = "paasta",
+) -> PrometheusAdapterRule:
+    """
+    Creates a Prometheus adapter rule config for a given service instance.
+    """
+    desired_active_requests_per_replica = autoscaling_config.get(
+        "desired_active_requests_per_replica", 1
```

Comment:
this `1` should probably be a constant (that we also use in validate.py) - otherwise, we're liable to have different "defaults" (which is the case right now :p)